src/python/pants/backend/python/goals/tailor.py

```diff
@@ -66,6 +96,62 @@ async def find_putative_targets(
                     tgt_type, dirname, name, sorted(filenames), kwargs=kwargs
                 )
             )
+
+    if python_setup.tailor_pex_binary_targets:
+        # Find binary targets.
+
+        # Get all files whose content indicates that they are entry points.
+        digest_contents = await Get(DigestContents, PathGlobs, all_py_files_globs)
+        entry_points = [
+            file_content.path
+            for file_content in digest_contents
+            if is_entry_point(file_content.content)
+        ]
+
+        # Get the modules for these entry points.
+        src_roots = await Get(
+            SourceRootsResult, SourceRootsRequest, SourceRootsRequest.for_files(entry_points)
+        )
+        module_to_entry_point = {}
+        for entry_point in entry_points:
+            entry_point_path = PurePath(entry_point)
+            src_root = src_roots.path_to_root[entry_point_path]
+            stripped_entry_point = entry_point_path.relative_to(src_root.path)
+            module = PythonModule.create_from_stripped_path(stripped_entry_point)
+            module_to_entry_point[module.module] = entry_point
+
+        # Get existing binary targets for these entry points.
+        entry_point_dirs = {os.path.dirname(entry_point) for entry_point in entry_points}
+        possible_existing_binary_targets = await Get(
+            UnexpandedTargets, AddressSpecs(AscendantAddresses(d) for d in entry_point_dirs)
+        )
+        possible_existing_binary_entry_points = await MultiGet(
+            Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest(t[PexEntryPointField]))
+            for t in possible_existing_binary_targets
+            if t.has_field(PexEntryPointField)
+        )
+        possible_existing_entry_point_modules = {
+            rep.val.module for rep in possible_existing_binary_entry_points if rep.val
+        }
+        unowned_entry_point_modules = (
+            module_to_entry_point.keys() - possible_existing_entry_point_modules
+        )
+
+        # Generate new targets for entry points that don't already have one.
+        for entry_point_module in unowned_entry_point_modules:
+            entry_point = module_to_entry_point[entry_point_module]
+            path, fname = os.path.split(entry_point)
+            name = os.path.splitext(fname)[0]
+            pts.append(
+                PutativeTarget.for_target_type(
+                    target_type=PexBinary,
+                    path=path,
+                    name=name,
+                    triggering_sources=tuple(),
+                    kwargs={"name": name, "entry_point": fname},
```

Comment:
This requires a sibling `python_library()` to work still IIRC. If that's right, is there more to do here or will that target already be tailored by other logic anyhow?

Reply:
Yes, but that sibling will already be tailored by other logic.